src/e2e/java/teammates/e2e/pageobjects/UserNotificationsPage.java

```diff
@@ -0,0 +1,123 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+
+/**
+ * Page Object Model for user notifications page.
+ */
+public class UserNotificationsPage extends AppPage {
+
+    @FindBy(id = "notification-tabs")
+    private WebElement notificationTabs;
+
+    @FindBy(id = "notifications-timezone")
+    private WebElement notificationsTimezone;
+
+    public UserNotificationsPage(Browser browser) {
+        super(browser);
+    }
+
+    @Override
+    protected boolean containsExpectedPageContents() {
+        return getPageSource().contains("Notifications");
+    }
+
+    public void verifyNotShownNotifications(NotificationAttributes[] notifications) {
+        List<String> shownNotificationIds = notificationTabs.findElements(By.className("card"))
+                .stream().map(e -> e.getAttribute("id")).collect(Collectors.toList());
+        for (NotificationAttributes notification : notifications) {
+            assertFalse(shownNotificationIds.contains(notification.getNotificationId()));
+        }
+    }
+
+    public void verifyShownNotifications(NotificationAttributes[] notifications, Set<String> readNotificationIds) {
+        // Only validates that the preset notifications are present instead of checking every notification
+        // This is because the page will display all active notifications in the database, which is not predictable
+        for (NotificationAttributes notification : notifications) {
+            verifyNotificationTab(notification, readNotificationIds);
+        }
+    }
+
+    public void verifyNotificationTab(NotificationAttributes notification, Set<String> readNotificationIds) {
+        boolean isRead = readNotificationIds.contains(notification.getNotificationId());
+        WebElement notificationTab = notificationTabs.findElement(By.id(notification.getNotificationId()));
+
+        // Check text and style of notification header
+        WebElement cardHeader = notificationTab.findElement(By.className("card-header"));
+        assertEquals(getHeaderText(notification), cardHeader.getText());
+        assertTrue(cardHeader.getAttribute("class").contains(getHeaderClass(notification.getStyle())));
+
+        // Checks if tab is open if notification is unread, and closed if notification is read
+        String chevronClass = notificationTab.findElement(By.tagName("i")).getAttribute("class");
+        if (isRead) {
+            assertTrue(chevronClass.contains("fa-chevron-down"));
+            // Open tab if notification is unread
+            click(cardHeader);
+            waitForPageToLoad();
```

Comment:
I believe `waitForPageLoad()` waits for the animation to finish, so `waitUntilAnimationFinish` can be removed